### PR TITLE
Goshko Volodymyr hw6

### PR DESCRIPTION
**Pull request by Ant3000**

### CODE REVIEW

File: GOshko/Ghw5/GoshkoHW5Task1.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import random
+xNumber = int(random.random() * 100)
+i = 0
+while i < 10:
+    xInput = int(input('Enter a integer number : '))
+    if xInput < xNumber:
+        print('Its bigger')
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task1.py:7]: Do you know how to use ' sign here?

File: GOshko/Ghw5/GoshkoHW5Task1.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    elif xInput > xNumber:
+        print('Its lower')
+    else:
+        print('!!! You guessed the number !!! ')
+        break
+    i += 1
+    if i == 10:
+        print('you didn\'t guess the number =(')
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task1.py:15]: Good Decision.

File: GOshko/Ghw5/GoshkoHW5Task2.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+div_can = 0
+for i in range(1, 11):
+    i = str(i)
+    num = int(input('Enter natural number, № ' + i + ' : '))
+    if num % 5 == 0:
+        div_can += 1
+else:
+    print(div_can, 'numbers can be divided into 5')
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task2.py:8]: Good Decision

File: GOshko/Ghw5/GoshkoHW5Task3.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+res_mul = 0
+for i in range(1, 10):
+    L = ' | '
+    print(i, '* |', end='  ')
+    for j in range(1, 10):
+        res_mul = j * i
+        if L == ' | ' and res_mul > 9:
+            L = '| '
+        print(res_mul, L, end=' ')
+    print()
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task3.py:12]: Nice Table

File: GOshko/Ghw5/GoshkoHW5Task4.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+if y < 0:
+    y = 0
+lineOut = " -" * x
+lineIn = " #" * (x - 2)
+print(lineOut)
+while y != 0:
+    print(' |' + lineIn + ' |')
+    y -= 1
+print(lineOut)
+# is it bad that I deviated from the task condition ???
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task4.py:12]: It's not bad. Nice picture.

File: GOshko/Ghw5/GoshkoHW5Task5.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+p = int(input('Enter min P : '))
+h = int(input('Enter max H : '))
+num = int(input('Enter some number : '))
+n_sum = 0
+while p != num != h:
+    if p < num < h:
+        n_sum += num
+    num = int(input('Enter some number : '))
+else:
+    print(n_sum)
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task5.py:10]: End what about two other tasks?

File: GOshko/Ghw5/GoshkoHW5Task6.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+while num != 0:
+    if num > 0:
+        n_pos += 1
+    else:
+        n_neg += 1
+    num = int(input('Enter a number : '))
+else:
+    num = n_pos + n_neg
+    print('percentage of positive number is', (n_pos / num) * 100)
+    print('percentage of negative number is', (n_neg / num) * 100)
```
Comment (katemalash) [inline GOshko/Ghw5/GoshkoHW5Task6.py:12]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task01.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    x = input('Enter a number : ')
+x = len(lineInp)
+i = 0
+while i != x:
+    lineRam.append(int(random.randrange(100)))
+    lineSum.append(lineRam[i] + lineInp[i])
+    i += 1
+print('numbers imputed :', lineInp)
+print('random numbers(0-99) :', lineRam)
+print('sum numbers:', lineSum)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task01.py:16]: Nice.

File: GOshko/Ghw6/GoshkoHW6Task10.py [hunk 9]
```diff
@@ -0,0 +1,6 @@
+names = ("ann", "Bob", "Elizabeth", "Mr. McMullen", "Mrs. Smith")
+for i in names:
+    if 'Mr' in i or 'Mrs' in i:
+        print('Good morning ' + i)
+    else:
+        print('Hello ' + i)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task10.py:6]: Good Decision

File: GOshko/Ghw6/GoshkoHW6Task09.py [hunk 10]
```diff
@@ -0,0 +1,8 @@
+tupleOne = (1, 2, 3, '#$', '@%', 'dace', 'girls', 'dog')
+listTuple = [str(i) for i in tupleOne]
+print('Tuple : ', tupleOne)
+ind = input('Enter one item from the tuple, enter 00 to exit : ')
+while ind != '00':
+    ind = listTuple.index(ind)
+    print('index of element is :', ind)
+    ind = input('Enter element form the tuple : ')
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task09.py:8]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task08.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
+dump = ('asd', 'asf33', 123, 234.5, True, False)
+number_line, string_line, bool_line, other_line = [], [], [], []
+for i in dump:
+    if int == type(i) or type(i) == float:
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task08.py:4]: it would be better to use `type(i) == int`

File: GOshko/Ghw6/GoshkoHW6Task08.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+        number_line.append(i)
+    elif type(i) == str:
+        string_line.append(i)
+    elif type(i) == bool:
+        bool_line.append(i)
+    else:
+        other_line.append(i)
+number_line, string_line, = tuple(number_line), tuple(string_line)
+bool_line, other_line = tuple(bool_line), tuple(other_line)
+print(number_line, string_line, bool_line, other_line)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task08.py:14]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task02.py [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    x = random.randrange(-5, 6)
+    list_ram.append(x)
+    if x > 0:
+        list_pos.append(x)
+    elif x < 0:
+        list_neg.append(x)
+else:
+    print('Random list is :', list_ram)
+    print('Positive numbers is :', list_pos)
+    print('Negative numbers is :', list_neg)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task02.py:14]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task03.py [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+len_list = int(input('Enter the length of the list : '))
+ram_list, index_list = [], []
+for i in range(len_list):
+    x = random.randrange(100)
+    ram_list.append(x)
+    if x%2 == 0:
+        index_list.append(i)
+else:
+    print('All numbers : ', ram_list)
+    print('Index of even numbers', index_list)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task03.py:11]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task04.py [hunk 15]
```diff
@@ -0,0 +1,17 @@
+import random
+columns = int(input('Enter the length of the columns : '))
+lines = int(input('Enter the length of the lines : '))
+matrix = []
+for i in range(0, columns):  # fill the matrix with numbers
+    matrix.append([])  # add empty list
+    for n in range(0, lines):
+        matrix[i].append(int(random.random() * 100))  # fill the matrix line
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task04.py:8]: Maybe `randrange(100)` would be better to use?

File: GOshko/Ghw6/GoshkoHW6Task04.py [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+matrix.append([])
+matrix[columns] = [0] * lines
+for i in range(0, columns):
+    matrix[i].append(0)
+    for n in range(0, lines):
+        matrix[i][-1] += matrix[i][n]
+        matrix[columns][n] += matrix[i][n]
+for i in matrix:
+    print(i)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task04.py:17]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task07.py [hunk 17]
```diff
@@ -0,0 +1,27 @@
+import random
+columns = int(input('Enter the length of the columns : '))
+lines = int(input('Enter the length of the lines : '))
+matrix, first_mtx, double = [], [], ' '
+for i in range(0, columns):  # fill the matrix with numbers
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task07.py:5]: is 0 required here?

File: GOshko/Ghw6/GoshkoHW6Task07.py [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+        x = first_mtx[0].index(matrix[0][j], x + 1, lines)
+        print(x)
+    be_sorted.append(x)
+    for i in range(1, columns):
+        matrix[i][j] = first_mtx[i][x]
+for i in first_mtx:
+    print(i)
+print()
+for i in matrix:
+    print(i)
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task07.py:27]: Good Decision.

File: GOshko/Ghw6/GoshkoHW6Task07.py [hunk 19]
```diff
@@ -0,0 +1,27 @@
+import random
+columns = int(input('Enter the length of the columns : '))
+lines = int(input('Enter the length of the lines : '))
+matrix, first_mtx, double = [], [], ' '
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task07.py:4]: What is double used for?

File: GOshko/Ghw6/GoshkoHW6Task05.py [hunk 20]
```diff
@@ -0,0 +1,19 @@
...
+for x in range(0, lines):
+    small = matrix[0][x]
+    for y in range(1, columns):
+        if small > matrix[y][x]:
+            small = matrix[y][x]
+    line_min.append(small)
+for i in matrix:
+    print(i)
+print('min num :', line_min)
+print('max num :', max(line_min))
```
Comment (katemalash) [inline GOshko/Ghw6/GoshkoHW6Task05.py:19]: Good Decision.

**Review by katemalash — APPROVED** [review]
Good Job. You can merge pull-request after fixing some issues.

File: GOshko/Ghw5/GoshkoHW5Task1.py (hunk 1 above)
Comment (Ant3000) [reply GOshko/Ghw5/GoshkoHW5Task1.py:7]: I do not understand this question

Comment (katemalash) [reply GOshko/Ghw5/GoshkoHW5Task1.py:7]: I mean `It's bigger` would be better.

File: GOshko/Ghw6/GoshkoHW6Task07.py (hunk 17 above)
Comment (Ant3000) [reply GOshko/Ghw6/GoshkoHW6Task07.py:5]: I still do not know why I write so =)))

File: GOshko/Ghw6/GoshkoHW6Task08.py (hunk 11 above)
Comment (Ant3000) [reply GOshko/Ghw6/GoshkoHW6Task08.py:4]: sorry, but I know that (int == type(i) == float) does not work =)))

File: GOshko/Ghw5/GoshkoHW5Task1.py (hunk 1 above)
Comment (Ant3000) [reply GOshko/Ghw5/GoshkoHW5Task1.py:7]: this is not a mistake in current English

**Pull request merged**